goSrc/src/vmCaller/iroha/storage.go

```diff
@@ -0,0 +1,159 @@
+package iroha
+
+// #cgo CFLAGS: -I ../../../../irohad
+// #cgo LDFLAGS: -Wl,-unresolved-symbols=ignore-all
+// #include <stdlib.h>
+// #include "ametsuchi/impl/burrow_storage.h"
+import "C"
+import (
+	"encoding/hex"
+	"unsafe"
+
+	"github.com/hyperledger/burrow/acm"
+	"github.com/hyperledger/burrow/binary"
+	"github.com/hyperledger/burrow/crypto"
+	"github.com/hyperledger/burrow/execution/errors"
+)
+
+type IrohaStorage struct {
+	storage unsafe.Pointer
+}
+
+func NewIrohaStorage(storage unsafe.Pointer) *IrohaStorage {
+	return &IrohaStorage{
+		storage: storage,
+	}
+}
+
+func handleIrohaCallResult(result C.Iroha_Result) (*string, error) {
+	switch result.which {
+	case C.Iroha_Result_Type_Value:
+		return result.data.toStringAndRelease(), nil
+	case C.Iroha_Result_Type_Error:
+		err_str := result.data.toStringAndRelease()
+		if err_str != nil {
+			return nil, errors.Errorf(errors.Codes.ExecutionReverted, *err_str)
+		}
+	}
+	return nil, errors.Errorf(errors.Codes.ExecutionReverted, "unknown error")
+}
+
+func (st *IrohaStorage) GetAccount(address crypto.Address) (*acm.Account, error) {
+	cAddress := MakeIrohaCharBuffer(address.String())
+	accountBytesHex, err := handleIrohaCallResult(C.Iroha_GetAccount(st.storage, *cAddress))
+	cAddress.free()
```

Comment:
Since both sides are represented by C interfaces, exceptions should be handled in C++ code and return as error code or something else to go through C interface. Same reasoning for go error handling.